pkg/cri/client/v1/client.go

```diff
@@ -0,0 +1,347 @@
+// Copyright Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+
+	v1 "k8s.io/cri-api/pkg/apis/runtime/v1"
+
+	logger "github.com/intel/cri-resource-manager/pkg/log"
+)
+
+type Client interface {
+	v1.ImageServiceClient
+	v1.RuntimeServiceClient
+}
+
+type client struct {
+	logger.Logger
+	isc v1.ImageServiceClient
+	rsc v1.RuntimeServiceClient
+	rcc *grpc.ClientConn
+	icc *grpc.ClientConn
+}
+
+// Connect v2alpha1 RuntimeService and ImageService clients.
+func Connect(runtime, image *grpc.ClientConn) (Client, error) {
+	c := &client{
+		Logger: logger.Get("cri/client"),
+		rcc:    runtime,
+		icc:    image,
+	}
+
+	if c.rcc != nil {
+		c.Info("probing CRI v1 RuntimeService client...")
+		c.rsc = v1.NewRuntimeServiceClient(c.rcc)
+		_, err := c.rsc.Version(context.Background(), &v1.VersionRequest{})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if c.icc != nil {
+		c.Info("probing CRI v1 ImageService client...")
+		c.isc = v1.NewImageServiceClient(c.icc)
+		_, err := c.isc.ListImages(context.Background(), &v1.ListImagesRequest{})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return c, nil
+}
+
+// HasRuntimeService checks if the client is configured with runtime services.
+func (c *client) HasRuntimeService() bool {
+	return c.rcc != nil
+}
+
+// HasImageService checks if the client is configured with image services.
+func (c *client) HasImageService() bool {
+	return c.icc != nil
+}
```

Comment:
`HasImageService()` is unused ATM, so I removed it. The different logic for HasRuntimeService in `pkg/client/client.go` is necessary (until we get rid of the `null` pseudo-policy), because it is use during startup to certain initialization actions. So it needs to consider configuration/desired state instead of the current state, which might be transiently different.